File: src/main/java/me/racci/sylphia/commands/ReloadManager.java
package me.racci.sylphia.commands;

import me.racci.sylphia.Sylphia;
import me.racci.sylphia.lang.CommandMessage;
import me.racci.sylphia.lang.Lang;
import me.racci.sylphia.lang.Prefix;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;

public class ReloadManager {

    private final Sylphia plugin;
    private final Lang lang;

    public ReloadManager(Sylphia plugin) {
        this.plugin = plugin;
        this.lang = plugin.getLang();
    }

    public void reload(CommandSender sender) {
        // Load config
        plugin.reloadConfig();
        plugin.saveDefaultConfig();
        plugin.getOptionLoader().loadOptions();
        // Load language files
        lang.load();
        lang.loadEmbeddedMessages(plugin.getCommandManager());
        lang.loadLanguages(plugin.getCommandManager());
        sender.sendMessage(Lang.getMessage(Prefix.SYLPHIA) + ChatColor.GREEN + Lang.getMessage(CommandMessage.RELOAD));
    }

}

File: src/main/kotlin/me/racci/sylphia/commands/ReloadManager.kt
@file:Suppress("unused")
@file:JvmName("ReloadManager")
package me.racci.sylphia.commands

import me.racci.raccilib.utils.strings.colour
import me.racci.sylphia.Sylphia
import me.racci.sylphia.lang.Command
import me.racci.sylphia.lang.Lang
import me.racci.sylphia.lang.Prefix
import org.bukkit.command.CommandSender

class ReloadManager(private val plugin: Sylphia) {
    fun reload(sender: CommandSender) {
        // Load config
        plugin.reloadConfig()
        plugin.saveDefaultConfig()
        plugin.optionLoader?.loadOptions()
        // Load language files
        plugin.lang?.loadLang(plugin.commandManager!!)
        sender.sendMessage(colour("${Lang.Messages.get(Prefix.SYLPHIA)} &8${Lang.Message.get(Command.RELOAD)}",false)!!)
    }
}